common/src/command.rs

```diff
@@ -17,13 +95,21 @@ impl<T: CommandType> CommandTypeDynamic for T {
 
 #[derive(Debug, Clone)]
 pub struct Speak(pub String);
```

Comment:
ahh, so it's like that, commands are structs? why don't we have a data carying enum for all commands, and perhaps a variant called custom or whatever for adons, if that's the problem?

Reply:
This allows handlers to take arguments like this:

```rust
async fn speak_handler(
    Speak(text): Speak, // text extracted from the `Command` enum; you can not use `Command::Speak(text)` if `Command::Speak(String)` is the data within the enum. We want to make this easy to do. The handler should not have to `try_into` on a generic command type.
    Speech(ssip): Speech, // extracted from state
) -> ... {
    
}
```

Reply:
Additionally, we can not extract a straight up `String`, because there may be other commands of which their only data contained is also a String. There would be no way to distinguish between the various discriminant of the enum.

Reply:
So you just want me to move this to its own fil? For command handlers?